src/treetensornetworks/projttns/projttn.jl

```diff
@@ -20,38 +21,83 @@ function set_nsite(P::ProjTTN, nsite)
 end
 
 function shift_position(P::ProjTTN, pos)
-  return ProjTTN(pos, P.H, P.environments)
+  return ProjTTN(pos, operator(P), environments(P))
 end
 
-function make_environment!(P::ProjTTN{V}, psi::TTN{V}, e::NamedEdge{V})::ITensor where {V}
+set_environments(p::ProjTTN, environments) = ProjTTN(pos(p), operator(p), environments)
+function set_environment(p::ProjTTN, edge, env)
+  newenvs = merge(environments(p), Dictionary((edge,), (env,)))
+  return ProjTTN(pos(p), operator(p), newenvs)
+end
```

Comment:
I we keep `set_environment!` (which is ok with me) it is better to implement this as:
```julia
set_environment(p::AbstractProjTTN, edge, env) = set_environment!(copy(p), edge, env)
```
to not have two implementations, i.e. it is a good practice to define out-of-place versions of a function int terms of in-place versions by copying the input.